File: packages/typia/src/transformers/features/IsTransformer.ts
import { IsProgrammer } from "@typia/core";

import { ITransformProps } from "../ITransformProps";
import { GenericTransformer } from "../internal/GenericTransformer";

export namespace IsTransformer {
  export const transform =
    (config: IsProgrammer.IConfig) => (props: ITransformProps) =>
      GenericTransformer.scalar({
        ...props,
        method: config.equals ? "equals" : "is",
        write: (x) =>
          IsProgrammer.write({
            ...x,
            config,
          }),
      });
}

File: packages/transform/src/features/IsTransformer.go
package features

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var IsTransformer = isTransformerNamespace{}

type isTransformerNamespace struct{}

func (isTransformerNamespace) Transform(equals bool) internal.Task {
	return internal.GenericTransformer.Scalar(programmers.IsProgrammer.Write)
}
